kyaml/yaml/walk/map.go

```diff
@@ -27,6 +27,9 @@ func (l Walker) walkMap() (*yaml.RNode, error) {
 
 	// recursively set the field values on the map
 	for _, key := range l.fieldNames() {
+		if key == "ports" {
```

Comment:
Although this hack can bypass the port merging issue, but `yaml.walk` is a very fundamental function in kyaml and kyaml is not only used by kustomize. Blindly skip `ports` field may have unpredictable results in some cases.

Reply:
Understood, I'll keep looking through the code over the next few days and try to fix it properly via schema and update the PR.